tpm2/tpm2.go

```diff
@@ -1491,3 +1491,92 @@ func encryptDecryptSymmetric(rw io.ReadWriteCloser, keyAuth string, key tpmutil.
 
 	return out, nil
 }
+
+func encodeRSAEncrypt(key tpmutil.Handle, message []byte, scheme *AsymScheme, label string) ([]byte, error) {
+	ha, err := tpmutil.Pack(key)
+	if err != nil {
+		return nil, err
+	}
+	m, err := tpmutil.Pack(message)
+	if err != nil {
+		return nil, err
+	}
+	s, err := scheme.encode()
+	if err != nil {
+		return nil, err
+	}
+	l, err := tpmutil.Pack([]byte(label))
+	if err != nil {
+		return nil, err
+	}
+	return concat(ha, m, s, l)
+}
+
+func decodeRSAEncrypt(resp []byte) ([]byte, error) {
+	var out []byte
+	_, err := tpmutil.Unpack(resp, &out)
+	return out, err
+}
+
+// RSAEncrypt performs RSA encryption in the TPM according to RFC 3447. The key must be
+// a (public) key loaded into the TPM beforehand. Note that when using OAEP with a label,
+// the label needs to be null-terminated and the null byte is included in the padding
+// scheme.
+func RSAEncrypt(rw io.ReadWriter, key tpmutil.Handle, message []byte, scheme *AsymScheme, label string) ([]byte, error) {
```

Comment:
What's the limit on message size? Is there a way to chain RSAEncrypt calls to encrypt a large blob?

Reply:
The size of the message that can be en-/decrypted depends on the size of the modulus, the algorithm used as well as the hash size. The random-looking numbers I used in the test represent the max for each combination, using a 2048 bit key. The formulas to calculate the max message size can be found in the spec, part 3, table 44. I'm happy to add those to a comment if you prefer.

As for use case, I imagine this can be used to encrypt a large blob with a symmetric key, then encrypt the symmetric key with the public RSA key (in software or a TPM). This key then can only be decrypted in the target TPM, with the blob then presumably decrypted in software using the symmetric key.